addons/arrays/fnc_prettyFormat.sqf

```diff
@@ -0,0 +1,71 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_prettyPrint
+
+Description:
+    Makes an array easier to read.
+
+Parameters:
+    _array - The array to format <ARRAY>
+    _tab   - The tab char to use (optional, default: tab) <STRING>
+    _depth - Starting indent (optional, default: 0) <NUMBER>
+
+Returns:
+    Formatted string <STRING>
+
+Examples:
+    (begin example)
+        _return = [ [0, 1, ["22", 33, []], 4] ] call CBA_fnc_prettyPrint;
+        // _return ==>
+        // "[
+        //      0,
+        //      1,
+        //      [
+        //          ""22"",
+        //          33,
+        //          []
+        //      ],
+        //      4
+        //  ]"
+        _return = [ [0, 1, ["22", 33, []], 4], ">---" ] call CBA_fnc_prettyPrint;
+        // _return ==>
+        // [
+        // >---0,
+        // >---1,
+        // >---[
+        // >--->---""22"",
+        // >--->---33,
+        // >--->---[]
+        // >---],
+        // >---4
+        // ]
+    (end)
+
+Author:
+   Terra
+---------------------------------------------------------------------------- */
+SCRIPT(prettyPrint);
+params [["_array", [], [[]]], ["_tab", toString [9], [""]], ["_depth", 0, [0]]];
+private _tabs = [];
+_tabs resize _depth;
+_tabs = _tabs apply {_tab} joinString "";
+if (_array isEqualTo []) exitWith {
+    _tabs + "[]"
+};
+private _lines = [_tabs + "["];
+private _lastIndex = count _array - 1;
+{
+    private _line = "";
```

Comment:
after changes we do have small performance hit with this useless assignment. @commy2 do you think it's worth it?

Reply:
I don't get what this is going for. The conditional below does `_line = _line + `, but `_line` is always the empty string?

Seems to me like below it should just be `_line =`. Put `private "_line";` above the loop to not leak it.